### PR TITLE
Feature/assignment 4

### PR DESCRIPTION
**Pull request by ayush1-successive**

### CODE REVIEW

File: Assignment-2/task-1.js
Comment (saurabh-yadav-successive) [inline Assignment-2/task-1.js:1]: it gives 0 also in result please correct it

File: Assignment-2/task-2.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Write a program to take input(age, name, phone no.) from user and display it
+
+const readline = require("readline");
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+rl.question("Enter name: ", (name) => {
```
Comment (saurabh-yadav-successive) [inline Assignment-2/task-2.js]: Please optimize code
Ex:-
create a separate function1 for questions that return promises. 
And then create a separate function2 which calls function1 and after resolving function1 log the result.

File: Assignment-2/task-3.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+//  Write a function expression that takes in another function as an argument
+
+function func() {
```
Comment (saurabh-yadav-successive) [inline Assignment-2/task-3.js]: Please explain the js concept that you are using in the solution.

File: Assignment-2/task-7.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// All of the above questions from 3-6 needs to be done with arrow functions also
+
+// Task-3
+
+const func = () => console.log("Hello World!");
+
+const takeAnotherFunc = (anotherFunc) => console.log(anotherFunc());
```
Comment (saurabh-yadav-successive) [inline Assignment-2/task-7.js]: log is undefined.please check

File: Assignment-2/task-7.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (saurabh-yadav-successive) [inline Assignment-2/task-7.js]: log is undefined. Pls check

File: Assignment-3/task-1.js
Comment (saurabh-yadav-successive) [inline Assignment-3/task-1.js:1]: Not working properly. Please add validations on input values. It takes alphanumeric values also.

**Review by saurabh-yadav-successive — COMMENTED** [review]
Please resolve comments

File: Assignment-2/task-2.js (hunk 1 above)
Comment (ayush1-successive) [reply Assignment-2/task-2.js]: I didn't consider using promises as they started from assignment-7. Will fix it.

**Review by saurabh-yadav-successive — APPROVED** [review]

**Pull request merged**